### PR TITLE
fix: reduce unnecessary k8s caching

### PR DESCRIPTION
**Pull request by g-linville**
This pretty substantially reduces the amount of memory usage from caching unnecessary Kubernetes information.

### CODE REVIEW

File: aggregator/data.go [hunk 1]
```diff
@@ -126,16 +126,8 @@ func (a *Aggregator) processk8s() {
...
-			a.processReplicaSet(d)
-		case k8s.DEPLOYMENT:
-			a.processDeployment(d)
-		case k8s.ENDPOINTS:
-			a.processEndpoints(d)
 		case k8s.CONTAINER:
 			a.processContainer(d)
-		case k8s.DAEMONSET:
-			a.processDaemonSet(d)
 		default:
```
Comment (iwilltry42) [inline aggregator/data.go:131]: I guess this will now log a lot of warnings, since you're removing a lot of cases?

**Review by iwilltry42 — APPROVED** [review]
If we won't need those, then :wave:

File: aggregator/data.go (hunk 1 above)
Comment (g-linville) [reply aggregator/data.go:131]: No, we aren't watching all of these removed resource types anyway, so we won't end up sending messages to the channel that this function is pulling from, so we shouldn't see these logs.

Comment (iwilltry42) [reply aggregator/data.go:131]: OH yeah, rereading this it makes total sense :+1:

**Review by tylerslaton — APPROVED** [review]

**Pull request merged**